### PR TITLE
WIP: Switch onImageRendered and onNewImage handlers to listen for native Custom Events

### PR DESCRIPTION
**Pull request by swederik**
This is the first step in https://github.com/chafey/cornerstone/issues/180

Please test the live examples and report anything you find broken:

https://rawgit.com/chafey/cornerstoneTools/custom-events/examples/index.html

### CODE REVIEW

**Comment by dannyrb** [comment]
**Issues:**
_Not sure if these are due to recent changes_

- Left click drag on **Freeform ROI** seems off. No visible change/active color until drag is released.
    - Similar issue here: https://rawgit.com/chafey/cornerstoneTools/custom-events/examples/freehandRoi/index.html
- Load Handler example throws error: https://rawgit.com/chafey/cornerstoneTools/custom-events/examples/loadHandlers/index.html
- I either don't know how this one is supposed to work, or I could not get it to work: https://rawgit.com/chafey/cornerstoneTools/custom-events/examples/stackImagePositionOffsetSynchronizer/index.html

I was able to verify touch tools using my laptop's touch screen :+1:
I'll take a look at the actual code commits and do some find-all'ing to see if anything was left behind.

Unrelated, but it appears a few tools handle their disable/enabled active/deactivated states differently.

**Comment by swederik** [comment]
Thanks! I am currently integrating @Maistho's changes from https://github.com/chafey/cornerstoneTools/pull/274 (literally just pushed a minute ago) so I will try to address those issues.

File: src/imageTools/arrowAnnotate.js [hunk 1]
```diff
@@ -340,6 +340,7 @@ function addNewMeasurementTouch (touchEventData) {
 }
 
 function doubleClickCallback (e, eventData) {
+  console.log('doubleClickCallback');
```
Comment (Maistho) [inline src/imageTools/arrowAnnotate.js]: Should probably be removed right?

**Comment by swederik** [comment]
Tested this and it all looks good. Thanks for everyone's help!

**Pull request merged**